Glows cdf attrs update

anamanica opened the pull request:
# Change Summary

## Overview
Changed glows l1a cdf variable attributes yaml file as requested by Marek and GLOWS team.

## New Dependencies
<!--List any new dependencies introduced by these changes-->
None

## New Files
<!--List each new file and add bullets to describe the purpose/function of the new file.
This should be high level, but enough detail for the reviewer to understand what purpose(s) the
code in the file is serving-->
None

## Deleted Files
<!--List each deleted file and add one or more bullets with the rationale for why the file is being deleted-->
None

## Updated Files
<!--List each updated file and add bullets to describe the purpose/function of the updates.
This should be high level, but enough detail for the reviewer to understand what purpose(s) the
updated code in the file is serving-->
- imap_glows_l1a_variable_attrs.yaml
   - Updated the information in comparison to the file Marek and the GLOWS team sent to us.
- glows_l1a.py
   - Updated some changed variable names.

## Testing
None (Already covered in test_glows_l1a.py)

maxinelasp committed:
first pass

anamanica committed:
MS mods

anamanica committed:
Glows MS CDF attrs update

anamanica committed:
Fixing tests

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
-# Info defined from glows_cdf_attrs
 int_fillval: &int_fillval -9223372036854775808
+int_uint32: &int_uint32 4294967295
+int_uint32_min_one: &int_uint32_min_one 4294967294
```
Rather than this, I would just use `*int_uint32 - 1` just because it seems a little overkill when `int_unit32 -1` makes a lot of sense, but that's a nitpick so feel free to ignore

anamanica replied:
No, I totally agree with you, it is quite redundant. Is there a way I can do this? Chat gpt informed me that .yaml files don't support operations like that (which could totally be wrong of course) 😅

maxinelasp replied:
I'm not sure, if it's not possible then it's fine how it is

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
 int_fillval: &int_fillval -9223372036854775808
+int_uint32: &int_uint32 4294967295
+int_uint32_min_one: &int_uint32_min_one 4294967294
+int_uint16: &int_uint16 65535
```
I also think all these variables should be `max_uint16` etc

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
-  # Resolution is optional?
   RESOLUTION: ' '
   DISPLAY_TYPE: no_plot
   TIME_BASE: J2000
```
Could you add a TODO to remove these unneeded attributes once SAMMI is fixed?

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
 
-# Information from glows_cdf_attrs.py
+epoch:
+  <<:  *default_attrs
```
Do we need default_attrs on this? Actually, Tenzin had a PR recently about epoch - can you look at that and see if we should use that style instead of manually defining it here?

anamanica replied:
Sure thing! It looks like we just need to remove the epoch variable dimension, and ImapCdfAttriutes will load in epoch from imap_constant_attrs.yaml.

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
-per_second_attrs:
+within_the_second: # Used to be per_second_attrs
   <<:  *default_attrs
   VALIDMMIN: 0
```
Still a typo here

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
-event_attrs:
+direct_event_components_attrs:
   <<:  *default_attrs
   VALIDMMIN: 0
```
typo

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
+direct_events:
   <<:  *default_attrs
   DEPEND_O: epoch
   DEPEND_1: per_second
```
Should be `within_the_second`

anamanica replied:
I thought that we wanted to change `per_second_attrs` to `within_the_second`, and `direct_event_attrs` to just `direct_events`. Is this wrong?

maxinelasp replied:
Ah, I meant this specific line - instead of `DEPEND_1: per_second` it should be `DEPEND_1: within_the_second` to point to our new dimension

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
   DISPLAY_TYPE: time_series
   LABLAXIS: Counts
-  FILL_VAL: *int_fillval
+  FILL_VAL: *int_uint16
```
We need to update the code to match this, did you do it or add a TODO comment?

anamanica replied:
How do we need to change the code to match this exactly?

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
-  # Can range from F0 to F0.9
+  CATDESC: Histogram start time, IMAP-clock seconds
+  FIELDNAM: Histogram start time, IMAP-clock seconds
+  FILLVAL: *int_fillval
```
Is this right? It is a float

maxinelasp replied:
Maybe -1?

anamanica replied:
It does appear that this is the number Marek wrote in the file you sent me, but there is a little comment next to it. I added a TODO here instead of changing it.

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
-  FORMAT: F16.6
+  CATDESC: Accumulation time in seconds for GLOWS histogram
+  FIELDNAM: Histogram accumulation time
+  FILLVAL: *int_fillval
```
Same comment

maxinelasp replied:
Actually, it might be good to make some kind of `time_defaults` for the shared bits here

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
+  FILLVAL: *int_uint32
+  FORMAT: I6
+  LABLAXIS: Mask value
+  VALIDMAX: *int_uint16
```
Max is uint16, fillval is uint32 - is this what they had in their doc? that seems weird to me. Could you add a todo to validate that with GLOWS?

anamanica replied:
Yeah, I just double checked and that is in the document. I added that comment!

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
+  VALIDMIN: 1
 
-number_of_spins_per_histogram:
+number_of_spins_per_histogram: # Should this be removed?
```
If this isn't used anywhere it should be removed, I think it's duplicated from number_of_bins_per_histogram

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
   UNITS: counts
   VAR_TYPE: data
 
 metadata_attrs: &metadata_attrs
```
can you a. move this up to the rest of the defaults and b. rename it to `support_data_defaults` to be more clear?

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
+  FIELDNAM: Variance of filter temperature
+  FILLVAL: *int_uint32
+  FORMAT: I6
+  LABLAXIS: Variance
```
Can you change this to "Tmp Variance?"

anamanica committed:
Updating Branch

Merge branch 'dev' of github.com:IMAP-Science-Operations-Center/imap_processing into glows_cdf_attrs_update

maxinelasp commented on imap_processing/cdf/config/imap_glows_l1a_variable_attrs.yaml:
```diff
@@ -1,358 +1,523 @@
+  FILLVAL: *int_uint32
+  FORMAT: I6
+  LABLAXIS: Voltage
+  VALIDMAX: 65635 # Is this a typo?
```
I think this should should be the same as the rest of the UINT values. It might be worth pulling them out as a default grouping, but you already have them in each specific variable so it's up to you if that's worth doing

maxinelasp commented on imap_processing/glows/l1a/glows_l1a.py:
```diff
@@ -258,7 +260,7 @@ def generate_de_dataset(
             "per_second": per_second,
```
per_second should be within_the_second here too

maxinelasp reviewed (commented):
Sorry for the delay in reviewing, I forgot 😅 some minor requested changes from me. Thanks for doing this!

anamanica committed:
Updating Branch

Merge branch 'dev' of github.com:IMAP-Science-Operations-Center/imap_processing into glows_cdf_attrs_update

anamanica committed:
Using default epoch

anamanica committed:
All PR comments gone through

anamanica committed:
Updating Branch

Merge branch 'dev' of github.com:IMAP-Science-Operations-Center/imap_processing into glows_cdf_attrs_update

maxinelasp commented on imap_processing/glows/l1a/glows_l1a.py:
```diff
@@ -244,20 +244,20 @@ def generate_de_dataset(
         direct_events,
         name="direct_events",
-        dims=["epoch", "per_second", "direct_event"],
+        dims=["epoch", "within_the_second", "direct_event"],
```
So since this is updated here, the `DEPEND_n` values that depend on this coordinate also need to be updated to match.

maxinelasp reviewed (commented):
One minor change but otherwise this is looking pretty good!

anamanica committed:
Updating branch.

Merge branch 'dev' of github.com:IMAP-Science-Operations-Center/imap_processing into glows_cdf_attrs_update

anamanica committed:
Depend_1 attr change

maxinelasp reviewed (approved):
Looks good, thanks for doing this!
